.github/workflows/terraform-plan.yml

```
@@ -0,0 +1,153 @@
+name: Terraform Plan
+
+on:
+  workflow_call:
+    
+    inputs:
+      environment:
+        required: true
+        type: string
+      plan_args:
+        required: false
+        type: string
+        description: Arguments provided to the plan command, e.g. terraform plan -var-file=dev.tfvars
+      terraform_dir:
+        required: false
+        type: string
+        default: terraform
+        description: Path to directory, where terraform is situated relative to the repo root. Should not have trailing slash.
+      plan_output_artifact:
+        required: false
+        type: boolean
+        default: true
+        description: Upload an artifact with the name "<environment>_plan", which can be used in the apply workflow to save execution time
+      assume_profile_arn:
+        required: false
+        type: string
+        description: If provided, this profile will be assumed before executing the terraform.
+      comment_to_pr:
+        required: false
+        type: boolean
+        default: true
+      
+    secrets:
+      AWS_ACCESS_KEY_ID:
+        required: true
+      AWS_SECRET_ACCESS_KEY:
+        required: true
+      TERRAFORM_CLOUD:
+        required: true
+      SSH_KEY:
+        required: false
+
+jobs:
+  terraform-plan:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ${{ inputs.terraform_dir }}
+    permissions: 
+      pull-requests: write
+      contents: read
+    steps:
+    - uses: actions/checkout@v3
+    - uses: hashicorp/setup-terraform@v2
+
+    - name: Configure AWS credentials
+      uses: aws-actions/configure-aws-credentials@v1
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY}}
+        aws-region: eu-west-1
+        role-to-assume: ${{ inputs.assume_profile_arn }}
+        role-duration-seconds: 2400
+
+    - name: Set up SSH
+      uses: webfactory/ssh-agent@v0.5.4
```

```suggestion
      uses: webfactory/ssh-agent@v0.6.0
```
Upgrade to node16 version to solve warnings